pkg/client/client_test.go

```diff
@@ -648,23 +651,184 @@ var _ = Describe("Client", func() {
+					},
+				},
+			}
+			depBackend, err = clientset.AppsV1().Deployments(ns).Create(depBackend)
+			Expect(err).NotTo(HaveOccurred())
 
-		It("should filter results by field selector", func() {
+			cl, err := client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
 
-		})
+			By("listing all Deployments with label app=backend")
+			deps := &appsv1.DeploymentList{}
+			labels := map[string]string{"app": "backend"}
+			lo := client.MatchingLabels(labels)
+			Expect(cl.List(context.Background(), lo, deps)).NotTo(HaveOccurred())
+
+			By("only the Deployment with the backend label is returned")
+			Expect(deps.Items).NotTo(BeEmpty())
+			Expect(1).To(Equal(len(deps.Items)))
+			actual := deps.Items[0]
+			Expect(actual.Name).To(Equal("deployment-backend"))
+
+			deleteDeployment(depFrontend, ns)
+			deleteDeployment(depBackend, ns)
+
+			close(done)
+		}, serverSideTimeoutSeconds)
+
+		It("should filter results by namespace selector", func(done Done) {
+			By("creating a Deployment in test-namespace-1")
+			depFrontend := &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "deployment-frontend", Namespace: "test-namespace-1"},
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"app": "frontend"},
+					},
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "frontend"}},
+						Spec:       corev1.PodSpec{Containers: []corev1.Container{{Name: "nginx", Image: "nginx"}}},
+					},
+				},
+			}
+			depFrontend, err := clientset.AppsV1().Deployments("test-namespace-1").Create(depFrontend)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating a Deployment in test-namespace-2")
+			depBackend := &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "deployment-backend", Namespace: "test-namespace-2"},
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"app": "backend"},
+					},
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "backend"}},
+						Spec:       corev1.PodSpec{Containers: []corev1.Container{{Name: "nginx", Image: "nginx"}}},
+					},
+				},
+			}
+			depBackend, err = clientset.AppsV1().Deployments("test-namespace-2").Create(depBackend)
+			Expect(err).NotTo(HaveOccurred())
+
+			cl, err := client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("listing all Deployments in test-namespace-1")
+			deps := &appsv1.DeploymentList{}
+			lo := client.InNamespace("test-namespace-1")
+			Expect(cl.List(context.Background(), lo, deps)).NotTo(HaveOccurred())
+
+			By("only the Deployment in test-namespace-1 is returned")
+			Expect(deps.Items).NotTo(BeEmpty())
+			Expect(1).To(Equal(len(deps.Items)))
+			actual := deps.Items[0]
+			Expect(actual.Name).To(Equal("deployment-frontend"))
+
```

Comment:
would be great if we can can another List() call here for `all-namespaces` and we get both the deployments. That will cover all the namespace based filtering.